### PR TITLE
Fixed Java Deployment

### PR DESCRIPTION
**Pull request by matrixjnr**
**What does this PR do?**

Fixed Java deployment logic and consequent builds that call `maven clean or gradle clean` to clear previous builds.

### CODE REVIEW

File: piku.py [hunk 1]
```diff
@@ -365,73 +365,61 @@ def do_deploy(app, deltas={}, newrev=None):
...
+
+    if not exists(build_path):
+        echo("-----> Building Java Application")
+        call('gradle build', cwd=join(APP_ROOT, app), env=env, shell=True)
+
     else:
-            if getmtime(build) > getmtime(target_path):
-                echo ("-----> Performing a clean build")
-                call('gradle clean build', cwd=join(APP_ROOT, app), env=env, shell=True)
+        echo("-----> Destroying previous builds")
```
Comment (dwightmulcahy) [inline piku.py]: "removing" vs. "Destroying" please

File: piku.py [hunk 2]
```diff
@@ -365,73 +365,61 @@ def do_deploy(app, deltas={}, newrev=None):
...
+        call('mvn package', cwd=join(APP_ROOT, app), env=env, shell=True)
+
     else:
-        if getmtime(pom) > getmtime(target_path):
-            
-            echo("-----> Destroying previous builds")
-            call('mvn clean', cwd=join(APP_ROOT, app), env=env, shell=True)
-            echo('-----> Starting new build')
-            call('mvn compile && mvn test && mvn package && mvn verify && mvn install && mvn deploy', cwd=join(APP_ROOT, app), env=env, shell=True)
+        echo("-----> Destroying previous builds")
```
Comment (dwightmulcahy) [inline piku.py]: see previous comment

**Comment by rcarmo** [comment]
Guess what, I'm not going to have the time to setup CI this weekend. @matrixjnr, @chr15m, could you please go over the stuff you think needs doing as follow-up and file issues?

**Comment by chr15m** [comment]
@rcarmo @matrixjnr I think the list looks like this:

 * PR for breaking out the old Java example from #99.
 * PR for the help & doc edits @matrixjnr made in #99.
 * PR for the Clojure lein builds from #99.

I'll try to make a start on these myself.

**Comment by matrixjnr** [comment]
> @rcarmo @matrixjnr I think the list looks like this:
> 
> * PR for breaking out the old Java example from #99.
> * PR for the help & doc edits @matrixjnr made in #99.
> * PR for the Clojure lein builds from #99.
> 
> I'll try to make a start on these myself.

I will start on Clojure builds

**Pull request merged**